training_main.py

```diff
@@ -0,0 +1,354 @@
+# %%
+
+# Importing generic python libraries
+import string
+import os
+import math
+import glob
+import numpy as np
+from numpy import array
+import pandas as pd
+import matplotlib.pyplot as plt
+get_ipython().run_line_magic('matplotlib', 'inline')
+from time import time
+
+# Importing libraries for image manipulation, deep-learning and pickling
+from PIL import Image, ImageOps
+from pickle import dump, load
+import tensorflow as tf
+
+# Importing functionalities from 'keras' library
+from keras.preprocessing import sequence
+from keras.models import Sequential
+from keras.layers import LSTM, Embedding, TimeDistributed, Dense, RepeatVector, Activation, Flatten, Reshape, concatenate, Dropout, BatchNormalization
+from keras.optimizers import Adam, RMSprop
+from keras.layers.wrappers import Bidirectional
+from keras.layers.merge import add
+from keras.applications.inception_v3 import InceptionV3
+from keras.preprocessing import image
+from keras.models import Model
+from keras import Input, layers
+from keras import optimizers
+from keras.applications.inception_v3 import preprocess_input
+from keras.preprocessing.text import Tokenizer
+from keras.preprocessing.sequence import pad_sequences
+from keras.utils import to_categorical
+
+#%%
+
+# Importing custom modules
+
+from utils import image_processing
+from utils import caption_writer
+from training_modules import description_processor
+from training_modules import description_properties
+from training_modules import training_functions
+
+# %%
+
+# Loading training descriptions into 'descriptions' dictionary
+
```

Comment:
you can load the descriptions and do their preprocessing in  other file and just call the final descriptions in this file.